runtime/common/src/main/kotlin/kotlinx/serialization/context/Context.kt

```diff
@@ -3,45 +3,175 @@
 package kotlinx.serialization.context
 
 import kotlinx.serialization.*
+import kotlinx.serialization.internal.*
 import kotlin.reflect.KClass
 
+private typealias SerializersMap = MutableMap<KClass<*>, KSerializer<*>>
+
+/**
+ * Serial context is a runtime mechanism used by [ContextSerializer] and [PolymorphicSerializer]
+ * to obtain serializers which were not found at compile-time by the serialization plugin.
+ *
+ * It can be regarded as a map where serializers are found using statically known KClasses.
+ */
 interface SerialContext {
+
+    /**
+     * Returns a dependent serializer associated with given [kclass].
+     *
+     * This method is used in context-sensitive operations
+     * on a property marked with [ContextualSerialization], by a [ContextSerializer]
+     */
     operator fun <T: Any> get(kclass: KClass<T>): KSerializer<T>?
 
+    /**
+     * Returns a serializer associated with KClass which given [value] has.
+     *
+     * This method is used in context-sensitive operations
+     * on a property marked with [ContextualSerialization], by a [ContextSerializer]
+     */
     fun <T: Any> getByValue(value: T): KSerializer<T>?
```

Comment:
should it really be part of the interface which everybody has to implement?
It can be used in-place via `get(value::class)`, why expose it in the base interface?